Documentation/Evolution/StringProcessingAlgorithms.md

```diff
@@ -574,27 +574,65 @@ extension RangeReplaceableCollection where SubSequence == Substring {
 ```swift
 extension Collection where Element: Equatable {
     /// Returns the longest possible subsequences of the collection, in order,
-    /// around elements equal to the given separator.
-    /// - Parameter separator: The element to be split upon.
+    /// around elements equal to the given separator collection.
+    ///
+    /// - Parameters:
+    ///   - separator: A collection of elements to be split upon.
+    ///   - maxSplits: The maximum number of times to split the collection, 
+    ///     or one less than the number of subsequences to return.
+    ///   - omittingEmptySubsequences: If `false`, an empty subsequence is 
+    ///     returned in the result for each consecutive pair of separator 
+    ///     sequences in the collection and for each instance of separator 
+    ///     sequences at the start or end of the collection. If `true`, only
+    ///     nonempty subsequences are returned.
     /// - Returns: A collection of subsequences, split from this collection's
-    /// elements.
-    public func split<S: Sequence>(by separator: S) -> some Collection<SubSequence>
-        where S.Element == Element
+    ///   elements.
+    public func split<C: Collection>(
+        by separator: C,
+        maxSplits: Int = Int.max,
+        omittingEmptySubsequences: Bool = true
+    ) -> some Collection<SubSequence> where C.Element == Element
 }
 
 extension BidirectionalCollection where SubSequence == Substring {
     /// Returns the longest possible subsequences of the collection, in order,
-    /// around elements equal to the given separator.
-    /// - Parameter separator: A regex describing elements to be split upon.
+    /// around subsequence that match the given separator regex.
+    ///
+    /// - Parameters:
+    ///   - separator: A regex to be split upon.
+    ///   - maxSplits: The maximum number of times to split the collection, 
+    ///     or one less than the number of subsequences to return.
+    ///   - omittingEmptySubsequences: If `false`, an empty subsequence is 
+    ///     returned in the result for each consecutive pair of matches
+    ///     and for each match at the start or end of the collection. If
+    ///     `true`, only nonempty subsequences are returned.
     /// - Returns: A collection of substrings, split from this collection's
-    /// elements.
-    public func split<R: RegexComponent>(by separator: R) -> some Collection<Substring>
+    ///   elements.
+    public func split(
+        by separator: some RegexComponent,
```

Comment:
Do we want to drop `by` and call it 
```swift 
public func split(separator... 
```

That would match the existing `String.split`

Reply:
I'm a little nervous about some of the overloads we already have in the stdlib — let me take a quick look.

Reply:
Okay, this should be safe — when someone has a type-unconstrained function reference, they'll still get the current `split(separator:maxSplits:omittingEmptySubsequences:)` method, because it's more tightly specified (i.e., to `Character`, instead of a generic collection). I don't _think_ adopting the lightweight generics syntax will change the story here.

Reply:
Even if this is from `_StringProcessing` instead of `Swift`?

Reply:
I added tests to make sure you still get the stdlib versions here: https://github.com/apple/swift-experimental-string-processing/pull/330/files#diff-3d2dc8728c9cb47deca496d7122ae9cd2e49118aa79f65f3b3c730c1adc51078R139-R143